tests/Integration/Behaviour/Features/Context/Domain/SecurityFeatureContext.php

```diff
@@ -0,0 +1,176 @@
+<?php
+/**
+ * Copyright since 2007 PrestaShop SA and Contributors
+ * PrestaShop is an International Registered Trademark & Property of PrestaShop SA
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.md.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://devdocs.prestashop.com/ for more information.
+ *
+ * @author    PrestaShop SA and Contributors <contact@prestashop.com>
+ * @copyright Since 2007 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ */
+
+declare(strict_types=1);
+
+namespace Tests\Integration\Behaviour\Features\Context\Domain;
+
+use Behat\Gherkin\Node\TableNode;
+use Context;
+use CustomerSession;
+use DateInterval;
+use DateTime;
+use Db;
+use EmployeeSession;
+use Exception;
+use PDO;
+use PHPUnit\Framework\Assert;
+use PrestaShop\PrestaShop\Core\Domain\Security\Command;
+use PrestaShop\PrestaShop\Core\Search\Filters\Security\Session;
+use RuntimeException;
+use Symfony\Component\HttpFoundation\Request;
+use Tests\Integration\Behaviour\Features\Context\SharedStorage;
+
+class SecurityFeatureContext extends AbstractDomainFeatureContext
+{
+    private const SECURITY_FORM_KEY = 'security-form';
+
+    /**
+     * @Given I specify following properties for security form
+     */
+    public function specifyPropertiesForSecurityForm(TableNode $node): void
+    {
+        $data = $node->getRowsHash();
+
+        SharedStorage::getStorage()->set(static::SECURITY_FORM_KEY, $data);
+    }
+
+    /**
+     * @When I submit the security form
+     */
+    public function submitTheSecurityForm(): void
+    {
+        $data = SharedStorage::getStorage()->get(static::SECURITY_FORM_KEY);
+
+        $request = Request::createFromGlobals();
+        $request->setMethod(Request::METHOD_POST);
+        $request->request->set(
+            'general',
+            $data
+        );
+
+        $formHandler = $this->getContainer()->get('prestashop.adapter.security.general.form_handler');
+        $form = $formHandler->getForm();
+        $form->handleRequest($request);
```

Comment:
Couldn't you use `$form->submit` instead here with data directly?
It would prevent "mocking" a request for not much interest

Reply:
FYI: This is what the handle request doing: 
https://github.com/symfony/symfony/blob/61cf298d192aae9cebdd8a342a6fd6db51243b98/src/Symfony/Component/Form/NativeRequestHandler.php#L124

Reply:
ok you convinced me